### PR TITLE
Implement "Plant trees" functionality

### LINKED ISSUE #9: Implement "Plant trees" functionality
## Scope

- Players emit CO2 on every trade
- A high global CO2 value, however, is not good for them though as trade yields become less
- Hence, some players might be incentivized to "plant trees" to combat the global CO2 pollution of the "Air"
- Planting a tree reduces the global CO2 value and the player's CO2 balance by X
- But of course, planting trees is not for free. It costs Goellars.
- In that sense, "Plant Trees" is the opposite of "Trade"

Inspiration (picture by @vrde)
image

## Deliverables

- Implement a simple "Plant trees" view that allows a player to offset their CO2 balance against Goellars
- Given the time constraints, it can really be as simple as adjusting a slider and clicking a "plant" button

## Roles
bounty gardener: @TimDaub / 10%
bounty worker: name / 75%
bounty reviewer: name / 15%

### PR DESCRIPTION
**Pull request by vrde**
Fix #9

### CODE REVIEW

File: src/planeta/PlantTrees.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      );
+    } catch (err) {
+      setReceipt({ type: "error", message: err.toString() });
+      changeView("receipt");
+      return;
+    }
+    setReceipt({
+      type: "plant",
+      txHash: result,
+      amount: amount * 16
```
Comment (TimDaub) [inline src/planeta/PlantTrees.js]: What is this `16` here?

File: src/planeta/PlantTrees.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    });
+    changeView("receipt");
+  }
+
+  render() {
+    const { balance, changeAlert, goBack } = this.props;
+    const { receipt, maxCO2 } = this.state;
+    console.log(maxCO2);
```
Comment (TimDaub) [inline src/planeta/PlantTrees.js]:
```suggestion
```

File: src/planeta/PlantTrees.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    // WARNING: handlePlantTrees accepts only integers, so no 0.1 Göllars or
+    // stuff like this. I had some troubles using BigInt with a floating point
+    // number.
+    return (
+      <Flex flexDirection="column">
+        <Heading.h5>Plant some trees</Heading.h5>
+        <Text fontSize={1}>Remove CO₂ from the atmosphere.</Text>
```
Comment (TimDaub) [inline src/planeta/PlantTrees.js:64]: wow such unicode

File: src/planeta/utils.js [hunk 4]
```diff
@@ -250,5 +269,96 @@ async function _finalizeHandshake(
...
+    .send(inputs, privateKey);
+}
+
+export async function maxCO2Available(plasma) {
+    return Math.max(...(await plasma.getUnspent(AirContractData.address, CO2_COLOR)).map(
+      ({ output: { value } }) => fromWei(value)
+    ).map(Number))
+}
+
+window.BN = BN;
```
Comment (TimDaub) [inline src/planeta/utils.js]: Uhm can we remove this? Was it just for debugging?

File: src/planeta/utils.js [hunk 5]
```diff
@@ -250,5 +269,96 @@ async function _finalizeHandshake(
...
+// 1 Göllar locks 16Gt of CO₂
+export async function plantTrees(plasma, passport, goellars, privateKey) {
+  const amount = new BN(CO2_PER_GOELLAR).mul(new BN(goellars));
+    console.log('asd222', amount.toString());
```
Comment (TimDaub) [inline src/planeta/utils.js]:
```suggestion
```

File: src/planeta/utils.js [hunk 6]
```diff
@@ -250,5 +269,96 @@ async function _finalizeHandshake(
...
+  const airContract = new PlasmaContract(plasma, AirContractData.abi);
+  const airLeapOutput = choice(
+    await plasma.getUnspent(AirContractData.address, LEAP_COLOR)
+  );
+  const airCO2Output = choice(
+    (await plasma.getUnspent(AirContractData.address, CO2_COLOR)).filter(
+      gt(amount.toString())
+    )
+  );
+  console.log(airLeapOutput, airCO2Output, amount.toString());
```
Comment (TimDaub) [inline src/planeta/utils.js:326]:
```suggestion
```

**Review by TimDaub — APPROVED** [review]
Just left a few comments but in general: LGTM

File: src/planeta/PlantTrees.js (hunk 3 above)
Comment (vrde) [reply src/planeta/PlantTrees.js:64]: ᵘⁿⁱᶜᵒᵈᵉ ⁱˢ ᶜᵒᵒˡ

**Review by TimDaub — APPROVED** [review]
LGTM

**Pull request merged**